apps/gdalmanage.cpp

```diff
@@ -92,73 +91,102 @@ static void ProcessIdentifyTarget(const char *pszTarget,
+                            "for the gdalmanage utility "
+                            "https://gdal.org/programs/gdalmanage.html"));
 
-    /* -------------------------------------------------------------------- */
-    /*      Process given files.                                            */
-    /* -------------------------------------------------------------------- */
-    for (; i < nArgc; ++i)
+    auto addCommonOptions = [psOptions](GDALArgumentParser *subParser)
     {
-        ProcessIdentifyTarget(papszArgv[i], nullptr, bRecursive,
-                              bReportFailures, bForceRecurse);
-    }
-}
+        subParser->add_argument("-f")
+            .metavar("<format>")
+            .store_into(psOptions->osDriverName)
+            .help(_("Specify format of raster file if unknown by the "
+                    "application."));
 
-/************************************************************************/
-/*                               Delete()                               */
-/************************************************************************/
+        subParser->add_argument("datasetname")
+            .metavar("<datasetname>")
+            .store_into(psOptions->osDatasetName)
+            .help(_("Name of the file to copy."));
 
-static void Delete(GDALDriverH hDriver, int nArgc, char **papszArgv)
+        subParser->add_argument("newdatasetname")
+            .metavar("<newdatasetname>")
+            .store_into(psOptions->osNewName)
+            .help(_("Name of the new file."));
+    };
 
-{
-    if (nArgc != 1)
-        Usage(true);
+    // Identify
 
-    GDALDeleteDataset(hDriver, papszArgv[0]);
-}
+    auto identifyParser =
+        argParser->add_subparser("identify", /* bForBinary */ true);
+    identifyParser->add_description(_("List data format of file."));
 
-/************************************************************************/
-/*                                Copy()                                */
-/************************************************************************/
+    identifyParser->add_argument("-r")
+        .flag()
+        .store_into(psOptions->bRecursive)
+        .help(_("Recursively scan files/folders for raster files."));
 
-static void Copy(GDALDriverH hDriver, int nArgc, char **papszArgv,
-                 const char *pszOperation)
+    identifyParser->add_argument("-fr")
+        .flag()
+        .store_into(psOptions->bRecursive)
+        .store_into(psOptions->bForceRecurse)
+        .help(_("Recursively scan folders for raster files, forcing "
+                "recursion in folders recognized as valid formats."));
 
-{
-    if (nArgc != 2)
-        Usage(true);
+    identifyParser->add_argument("-u")
+        .flag()
+        .store_into(psOptions->bReportFailures)
+        .help(_("Report failures if file type is unidentified."));
+
+    identifyParser->add_argument("datasetname")
+        .metavar("<datasetname>")
+        .store_into(psOptions->osDatasetName)
```

Comment:
identify allows several filenames to be specified

Reply:
done, plus I added tests and support multiple files for both identify and delete.